src/clr-angular/popover/signpost/signpost-trigger.ts

```diff
@@ -32,26 +33,37 @@ export class ClrSignpostTrigger implements OnDestroy {
   private subscriptions: Subscription[] = [];
   public ariaExpanded: boolean;
   public ariaControl: string;
+  private document: Document;
+  isOpen: boolean;
 
   constructor(
     private ifOpenService: IfOpenService,
-    private renderer: Renderer2,
     private el: ElementRef,
     public commonStrings: ClrCommonStringsService,
     private signpostIdService: SignpostIdService,
+    private signpostFocusManager: SignpostFocusManager,
+    private injector: Injector,
     @Inject(PLATFORM_ID) private platformId: Object
   ) {
+    this.document = this.injector.get(DOCUMENT);
+  }
+
+  ngOnInit() {
+    this.signpostFocusManager.triggerEl = this.el.nativeElement;
     this.subscriptions.push(
       this.ifOpenService.openChange.subscribe((isOpen: boolean) => {
-        if (isOpen) {
-          this.renderer.addClass(this.el.nativeElement, 'active');
+        this.ariaExpanded = isOpen;
+        // on initial value setting, we shouldn't move focus at all.
+        if (typeof this.isOpen === 'undefined' && typeof isOpen === 'boolean') {
+          this.isOpen = isOpen;
         } else {
-          this.renderer.removeClass(this.el.nativeElement, 'active');
-          if (isPlatformBrowser(this.platformId)) {
-            this.el.nativeElement.focus();
+          this.isOpen = isOpen;
```

Comment:
Yes, but...

You are setting `this.isOpen` to the same value whether the conditional is true or false.

What is the point of the conditional?

Reply:
Before assigning new `isOpen` emitted value that to `this.isOpen`, we need to save the previous value somewhere (in this, it's saved to the variable `prevIsOpen`)
```
const prevIsOpen = this.isOpen;
this.isOpen = isOpen; 
```
so we can check if previous isOpen is false and new isOpen is true.

Also, `this.isOpen` is public property used for binding.

Reply:
Mkay.

But that is not what this code is doing.

Here is the logic in this function...

```
if (x < 0) {
  y = 0;
} else {
  y = 0;
}

I don't see any logic here testing if the previous `this.isOpen` is true and the new one is `false`. I see logic that tests if the current value is undefined and the new one is a boolean.

And then setting the current value to the new value in BOTH cases.

Reply:
We first check if `this.isOpen` is `undefined`. If it's we will just set a new value to it. If it's no longer `undefined`, not only we still assign a new value, but also we call `this.signpostFocusManager.focusTrigger()`.  

But this part of the code has already been updated. I think the update makes it a bit clearer.

Reply:
Your update makes it clearer. The previous code wasn't doing what you describe. But the new code appears to be. Thanks.